test/controller_test.exs

```diff
@@ -3,61 +3,228 @@ defmodule ControllerTest do
   import Controller
   import ExUnit.CaptureIO
 
-  describe "parse_input/1" do
-    test "when passed a character on the menu, does not error" do
-      output = fn -> parse_input("V") end
-      assert capture_io(output) == "Which recipe would you like to view?\n1) Ice Cubes\n\n"
+  describe "run/1" do
+    test "when passed 'Q', prints the quit message and leaves the context map unchanged." do
+      quit_prompt = "\e[H\e[2JGoodbye!\n\n"
+      example_context = %{
+        input: "Q",
+        content: nil,
+        header: nil,
+        view: :index,
+        io: IO,
+        prompt: "Foo",
+        menu: "Bar"
+      }
+      output = fn -> run(example_context) end
+      assert capture_io(output) == quit_prompt
     end
   end
 
-  describe "execute_command/1" do
-    test "when passed 'V', prints a list of viewable recipes files" do
-      view_recipe_list = "1) Ice Cubes\n\n"
-      output = fn -> execute_command("V") end
-      assert capture_io(output) == view_recipe_list
-    end
+  describe "update_context/1" do
+    test "when view is :index, updates prompt, header and menu" do
+      example_context = %{
+        input: "I",
+        content: nil,
+        header: nil,
+        view: :index,
+        io: IO,
+        prompt: "Foo",
+        menu: "Bar"
+      }
 
-    test "when passed 'Q', prints the quit message." do
-      quit_prompt = "\e[H\e[2J\n\nGoodbye!\n\n\n"
-      output = fn -> execute_command("Q") end
-      assert capture_io(output) == quit_prompt
+      assert %{prompt: "Which recipe would you like to view? Type the number and press enter."} = update_context(example_context)
     end
+  end
 
-    test "when passed 'G', prints a grocery list." do
-      grocery_list =
-        "- 2 cups water (approximately)\n- 2 tablespoons water (additional if needed)\n\n"
+  describe "fetch_content/1" do
+      test "in view :welcome, when passed valid input 'I', changes view to :index and updates content" do
+        example_context = %{
+          input: "I",
+          content: nil,
+          header: nil,
+          view: :welcome,
+          io: IO,
+          prompt: "Foo",
+          menu: "Bar"
+        }
 
-      output = fn -> execute_command("G") end
-      assert capture_io(output) == grocery_list
-    end
+        assert %{content: ["1) Ice Cubes\n"]} = fetch_content(example_context)
+        assert %{view: :index} = fetch_content(example_context)
+      end
```

Comment:
I think it would be more readable if the tests were formatted a little bit differently. eg: 
```elixir
test "in view :welcome, when passed valid input 'I', changes view to :index and updates content" do
     # setup here
      example_context = %{
          input: "I",
          content: nil,
          header: nil,
          view: :welcome,
          io: IO,
          prompt: "Foo",
          menu: "Bar"
        }
      
      # call the function you are testing
      content =  fetch_content(example_context)

      # assertions
      assert %{content: ["1) Ice Cubes\n"]} = content
      assert %{view: :index} = content
    end
```

Reply:
Nice! Absolutely. Definitely clearer.